packages/ui/src/components/molecules/SummaryCard.tsx

```diff
@@ -0,0 +1,159 @@
+import React, { FC } from 'react';
+import styled, { useTheme } from 'styled-components';
+
+import { EditButton } from './Prefabs';
+
+import {
+  ClockIcon,
+  EmailIcon,
+  EncryptedMessageIcon,
+  MessageIcon,
+  SvgProps,
+  UserIcon,
+  WalletIcon,
+} from '../../assets';
+import { Flex, NomineeMessage, Divider, Typography } from '../atoms';
+import { svgGradients } from '../GlobalStyles';
+
+const iconMap: Record<string, React.FC<SvgProps>> = {
+  wallet: WalletIcon,
+  clock: ClockIcon,
+  user: UserIcon,
+  email: EmailIcon,
+  message: MessageIcon,
+  encrypted: EncryptedMessageIcon,
+};
+
+export type IconType = keyof typeof iconMap;
+export type BackgroundType = 'gold' | 'silver' | 'default';
+
+export interface SummaryField {
+  icon: IconType;
+  label: string;
+  value?: string;
+  onEdit?: () => void;
+  isDanger?: boolean;
+}
+
+export interface SummaryCardProps {
+  headerText: string;
+  headerIcon?: IconType;
```

Comment:
It does not need to be an svgComponent, since we are not handling any state change for headerIcon.

```suggestion
  headerIcon?: React.ReactNode;
```

Reply:
updated